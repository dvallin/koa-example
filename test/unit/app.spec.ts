import { testing } from '../test-modes'
import { MockPostgres } from '../test-modes/mock-database'
import { testMode, testManagementMode } from '../test-runners'
import * as request from 'supertest'

describe('app', () => {
  testMode(testing, 'registers migrations', async (_server, mode) => {
    const postgres = mode.components.io.postgres as MockPostgres
    expect(postgres.migrations).toMatchSnapshot()
  })

  testManagementMode(testing, 'becomes healthy', async server => {
    const response = await request(server).get('/instrumentation/health')
    expect(response.status).toBe(200)
    expect(response.text).toEqual('OK')
    expect(response.body).toEqual({})
  })

  testManagementMode(testing, 'becomes ready', async server => {
    const response = await request(server).get('/instrumentation/ready')
    expect(response.status).toBe(200)
    expect(response.text).toEqual('OK')
    expect(response.body).toEqual({})
  })

  testManagementMode(testing, 'becomes unready if postgres is disconnected', async (server, mode) => {
    const postgres = mode.components.io.postgres as MockPostgres
    postgres.isConnected = false

    const response = await request(server).get('/instrumentation/ready')

    expect(response.status).toBe(500)
    expect(response.text).toEqual('Internal Server Error')
    expect(response.body).toEqual({})
  })
})
