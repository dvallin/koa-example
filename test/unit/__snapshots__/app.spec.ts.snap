// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app with running server registers migrations 1`] = `
Array [
  Object {
    "text": "CREATE TABLE IF NOT EXISTS koa_users (
        email  VARCHAR(45) PRIMARY KEY,
        name   VARCHAR(45)
      );",
  },
  Object {
    "text": "CREATE TABLE IF NOT EXISTS chat_messages (
        room  VARCHAR(45),
        date  TIMESTAMP,
        message   TEXT,
        PRIMARY KEY(room, date)
      );",
  },
]
`;
